src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -137,25 +157,36 @@ where
 pub async fn convert_all_bits<F, C, S>(
     ctx: &C,
     locally_converted_bits: &[Vec<BitConversionTriple<S>>],
-) -> Result<Vec<Vec<S>>, Error>
+    num_bits: u32,
+    num_multi_bits: u32,
+) -> Result<Vec<Vec<Vec<S>>>, Error>
 where
     F: Field,
     C: Context<F, Share = S>,
     S: ArithmeticSecretSharing<F>,
 {
-    let futures = locally_converted_bits
-        .iter()
-        .enumerate()
-        .map(|(bit_num, one_column)| {
-            convert_bit_list(
-                ctx.narrow(&ModulusConversion(bit_num.try_into().unwrap()))
-                    .set_total_records(one_column.len()),
-                one_column,
-            )
-        })
-        .collect::<Vec<_>>();
-    let converted_shares = try_join_all(futures).await?;
-    Ok(converted_shares)
+    let num_records = locally_converted_bits.len();
+    try_join_all(
+        (0..num_bits)
+            .step_by(num_multi_bits.try_into().unwrap())
```

Comment:
`step_by` seems like not exactly what you want here. I think you want something more like `.chunks(num_multi_bits)` 

https://docs.rs/itertools/0.7.8/itertools/trait.Itertools.html#method.chunks

Looks like we use this a few places in our project already.

Reply:
chunks needs a vector to work on (and not a range) so needed to collect the bit nums before calling it. Let me know if it looks ok

Reply:
Oh, I see. Does calling "collect" result in instantiating a new vector?

Reply:
Yes it does